match/match.go

```diff
@@ -275,10 +276,11 @@ func registerEventHandlers(parser dem.Parser, match *Match) {
 }
 
 // parse demo and save GameStates in slice
-func parseGameStates(parser dem.Parser, match *Match) []common.OverviewState {
+func parseGameStates(parser dem.Parser, match *Match, pb *pb.ProgressBar) []common.OverviewState {
 	playbackFrames := parser.Header().PlaybackFrames
 	states := make([]common.OverviewState, 0, playbackFrames)
 
+	pb.SetTotal(pb.Total() + (int64(playbackFrames) / 4))
```

Comment:
I'm still not 100% sure why it needed to be divided by four for it to work properly 🤔
This wasn't a guess, btw. I started with the full playback frames and it would always finish loading and start the playback with 25% of the progress, so that is why I added this.

Reply:
We only take every 2nd or every 4th frame when parsing 64-tick or 128-tick demos, respectively. [link](https://github.com/Linus4/csgoverview/blob/66e29a9b329c30eda238d2b2bb59c4f6a470d574/match/match.go#L289)
I guess you were parsing a 128-tick demo, which is why the progress bar always stopped at 25 % :P The `match` type has a field called `takeNthFrame` which you can use in the division rather than `4`.

Reply:
Ooh, I see! I guess I missed that one 😆 But cool! Will update the PR